slyd/slyd/jsonapi/serializers.py

```diff
@@ -0,0 +1,655 @@
+        exclude = exclude_map.get(type_)
+
+        if not only:
+            f_set = fields and set(fields)
+            e_set = exclude and set(exclude)
+
+            self.field_set = field_set = set()
+            self.relationship_set = relationship_set = set()
+            for name in field_names:
+                if should_include_field(self.inflect(name),
+                                        include=f_set, exclude=e_set):
+                    field_set.add(name)
+            for name in relationship_names:
+                if should_include_field(self.inflect(name),
+                                        include=f_set, exclude=e_set):
+                    relationship_set.add(name)
+
+            only = list(field_set | {'id'})
+            for name in relationship_names:
+                relationship = self._declared_fields[name]
+                if (name in relationship_set or relationship.related_url or
+                        relationship.self_url):
+                    only.append(name)
+            only.append('_url')
+
+        self.field_order = list(
+            chain(fields or [], map(self.inflect, field_names)))
+        self.relationship_order = list(
+            chain(fields or [], map(self.inflect, relationship_names)))
+
+        only_set = set(only)
+        kwargs['include_data'] = tuple(
+            k for k in iterkeys(self.include_map) if k in only_set)
+        super(JsonApiSerializer, self).__init__(only=only, **kwargs)
+        self.instance = instance
+        self.initial_data = data
+        if storage:
+            self.storage = storage
+        elif isinstance(instance, Model):
+            self.storage = instance.storage
+        else:
+            self.storage = None
+        self.deleted = []
+        self.updated = []
+
+    @property
+    def data(self):
+        return self.dump(self.instance).data
+
+    @cached_property
+    def errors(self):
+        if not hasattr(self, '_errors'):
+            self.is_valid()
+        return self._errors
+
+    @cached_property
+    def validated_data(self):
+        if not hasattr(self, '_errors'):
+            self.is_valid()
+        return self._validated_data
+
+    @cached_property
+    def validated_profile_updates_data(self):
+        if not hasattr(self, '_errors'):
+            self.is_valid()
+        return self._validated_profile_updates_data
+
+    def deserialize_related_model(self, model, id_):
+        if id_ is None:
+            return None
+        return model(self.storage, **{
+            model._pk_field: id_,
+        })
+
+    def update(self, instance, validated_data):
+        model = instance.__class__
+
+        errors = {}
+        fields = []
+        for attrname in model._ordered_fields:
+            if attrname in validated_data:
+                try:
+                    value = validated_data[attrname]
+                    if attrname in model._field_names:
+                        setattr(instance, attrname, value)
```

Comment:
Perhaps a better way to model the interface for a model is to expose and `update` method which will store all the changes we wish to make. Once the `save` method is invoked, it will do a `save(only=fields)` call where the fields are the various `update` calls which were made before calling `save`.